### PR TITLE
Convert Github file link to raw content link

### PR DESCRIPTION
**Pull request by yongyan-gh**
Convert Github file link to raw content link

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs [hunk 1]
```diff
@@ -762,5 +762,32 @@ internal static bool OpenExternalUrl(string uriString, int maxLength = 150)
 
             return false;
         }
+
+        internal static string ConvertToGithubRawPath(string url)
```
Comment (michaelcfanning) [inline src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs:766]: >ConvertToGithubRawPath [](http://example.com/codeflow?start=31&length=22)

Why are we linking to raw git hub content? how does that help customers?

why aren't we jumping to the github hosted content with the appropriate line selected?

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

File: src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs (hunk 1 above)
Comment (yongyan-gh) [reply src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs:766]: this is the scenario that a result points to a github file url.
e.g.:
```
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "https://github.com/microsoft/sarif-visualstudio-extension/blob/main/src/Sarif.Viewer.VisualStudio/Data/ruleLookup.json"
                },
```
if send http request to this url directly (https://github.com/microsoft/sarif-visualstudio-extension/blob/main/src/Sarif.Viewer.VisualStudio/Data/ruleLookup.json), the response are html texts, its actually a html page includes a file content.
image

if only want to get the file content, the url need to be converted to raw github url e.g.:
https://raw.githubusercontent.com/microsoft/sarif-visualstudio-extension/main/src/Sarif.Viewer.VisualStudio/Data/ruleLookup.json
this url returns raw file content in plain text format.

In the extension when user navigates to a result which location is a github file link, it downloads the file content and saves it to local temp folder and opens the local file in VS editor. This change is to try to download the raw file content and not the html file content.

Comment (yongyan-gh) [reply src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs:766]: This is an example .sarif file can demonstrate the issue.
If you navigate to the result, it opens a file named applicationsetting.json but content is html text.

Comment (michaelcfanning) [reply src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs:766]: Thanks for this explanation. I understand now.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**